src/app/config-page/config-page.component.spec.ts

```diff
@@ -139,9 +140,69 @@ describe('ConfigPageComponent', () => {
     fixture = TestBed.createComponent(ConfigPageComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    configServiceStub = TestBed.get(ConfigService);
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should not save config changes if save is not pressed', done => {
+    fixture.detectChanges();
+    expect(component.displayedButtons).not.toBe(configServiceStub.displayedButtons);
+    const allTabs = fixture.nativeElement.querySelectorAll('div.mat-tab-label');
+    allTabs[2].click();
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+      fixture.detectChanges();
+      const showSpeakButton = fixture.debugElement.query(By.css('mat-checkbox label'));
```

Comment:
How does this get the showSpeakButton checkbox as opposed to the other checkboxes? Is it just the first checkbox? That would worry me as that's not a specific checkbox but you are accessing a specific attribute in the component to check against and reordering the checkboxes would break the tests even though the functionality is not broken...

Reply:
I've fixed it with some more xpath